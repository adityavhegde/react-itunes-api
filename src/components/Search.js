import React, {Component} from 'react';
import { Row, Col, FormControl } from 'react-bootstrap';

class Search extends Component {

  constructor(props) {
    super(props);
    this.searchHandler = (e) => {
      if(e.keyCode === 13) {
        let data = {query: document.getElementById("search-bar").value};
        this.props.queryCallback(data);
      }
    }
  }

  render() {
    return (
      <Row>
        <Col>
          <FormControl
            id="search-bar"
            type="text"
            placeholder="Search"
            onKeyUp={this.searchHandler}
          />
        </Col>
      </Row>
    )
  }
}

export default Search;
