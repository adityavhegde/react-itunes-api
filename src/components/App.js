import React, {Component} from 'react';
import Search from './Search';
import Header from './Header';
import ResultBody from './ResultBody';
import {Grid} from 'react-bootstrap';

class App extends Component {

  state = {
    searchText:''
  }

  updateSearchText(query) {
    this.setState({
      searchText:query
    });
  }

  render() {
    return (
      <Grid>
        <Header/>
        <Search queryCallback={this.updateSearchText.bind(this)}/>
        <br/>
        <ResultBody query={this.state.searchText}/>
      </Grid>
    )
  }
}

export default App;
