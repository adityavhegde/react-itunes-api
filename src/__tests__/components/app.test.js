import React from 'react';
import {shallow} from 'enzyme';
import App from '../../components/App';
import Search from '../../components/Search';

describe('<App />', () => {
  test('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
