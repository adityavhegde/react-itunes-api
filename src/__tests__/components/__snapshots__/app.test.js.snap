// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Header />,
        <Search
          queryCallback={[Function]}
/>,
        <br />,
        <ResultBody
          query=""
/>,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "queryCallback": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "query": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Header />,
          <Search
            queryCallback={[Function]}
/>,
          <br />,
          <ResultBody
            query=""
/>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "queryCallback": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "query": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
